optuna/samplers/_gp/sampler.py

```diff
@@ -214,42 +223,77 @@ def sample_relative(
+            kernel_params_list.append(
+                gp.fit_kernel_params(
+                    X=normalized_params,
+                    Y=standardized_score_vals[:, i],
+                    is_categorical=is_categorical,
+                    log_prior=self._log_prior,
+                    minimum_noise=self._minimum_noise,
+                    initial_kernel_params=cache,
+                    deterministic_objective=self._deterministic,
+                )
+            )
+        self._kernel_params_cache_list = kernel_params_list
 
         best_params: np.ndarray | None
         if self._constraints_func is None:
-            acqf_params = acqf.create_acqf_params(
-                acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
-                kernel_params=kernel_params,
-                search_space=internal_search_space,
-                X=normalized_params,
-                Y=standardized_score_vals,
-            )
-            best_params = normalized_params[np.argmax(standardized_score_vals), :]
+            if n_objectives == 1:
+                assert len(kernel_params_list) == 1
+                acqf_params = acqf.create_acqf_params(
+                    acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
+                    kernel_params=kernel_params_list[0],
+                    search_space=internal_search_space,
+                    X=normalized_params,
+                    Y=standardized_score_vals[:, 0],
+                )
+                best_params = normalized_params[np.argmax(standardized_score_vals), np.newaxis]
+            else:
+                acqf_params_for_objectives = []
+                for i in range(n_objectives):
+                    acqf_params_for_objectives.append(
+                        acqf.create_acqf_params(
+                            acqf_type=acqf.AcquisitionFunctionType.LOG_EHVI,
+                            kernel_params=kernel_params_list[i],
+                            search_space=internal_search_space,
+                            X=normalized_params,
+                            Y=standardized_score_vals[:, i],
+                        )
+                    )
+                acqf_params = acqf.MultiObjectiveAcquisitionFunctionParams.from_acqf_params(
+                    acqf_params_for_objectives=acqf_params_for_objectives,
+                    Y=standardized_score_vals,
+                    n_qmc_samples=128,  # NOTE(nabenabe): The BoTorch default value.
+                    qmc_seed=self._rng.rng.randint(1 << 30),
+                )
+                pareto_params = normalized_params[
+                    _is_pareto_front(standardized_score_vals, assume_unique_lexsorted=False)
+                ]
+                n_pareto_sols = len(pareto_params)
+                size = min(self._n_local_search // 2, n_pareto_sols)
+                chosen_indices = self._rng.rng.choice(
+                    np.arange(n_pareto_sols), size=size, replace=False
+                )
```

Comment:
If the solution on the Pareto-front is large enough, you randomly select `n_local_search // 2` solutions from it as `best_params`. Is there any reference for this selection?

Reply:
No, this is a heuristic inspired from the single-objective optimization by `GPSampler`.
We can also follow the BoTorch convention if necessary.

Reply:
I see. This raises a performance concern. In multi-objective optimization, unlike single-objective optimization, there are multiple Pareto-optimal solutions to reach (and in many cases, there are infinitely many). Therefore, we should ideally start from multiple points for each of these infinitely many Pareto solutions. Of course, that is not feasible, but at the very least, shouldn’t we be optimizing the acquisition function toward Pareto solutions using a larger number of initial points compared to single-objective optimization? I suspect that this might be the reason for the performance difference compared to BoTorch. How does BoTorch handle this?

Reply:
Could you check this comment?

Reply:
Although BoTorch itself does not have a fixed parameter for the number of local searches, `BoTorchSampler` does have it.
- Number of local search: 20 ([BoTorchSampler](https://github.com/optuna/optuna-integration/blob/v4.3.0/optuna_integration/botorch/botorch.py#L454)) vs 10 ([GPSampler](https://github.com/optuna/optuna/blob/v4.3.0/optuna/samplers/_gp/sampler.py#L141))

I would like you to refer to this comment for more details:
- https://github.com/optuna/optuna/pull/6069#issuecomment-2875949475